### DISCUSSION TITLE
Clarify tree shaking explanation from the docs

### DISCUSSION THREAD

**Discussion by yishay-at-bay** [webpack/webpack#18821] [Q&A]
The docs has a very detailed explanation about tree shaking and the sideEffects flag in package json
But not all of it is clear to me, especially [this part](https://webpack.js.org/guides/tree-shaking/#clarifying-tree-shaking-and-sideeffects)

Can someone please clarify the Shopify example or give similar example with more details about what code exists on each file and how the tree shaking logic works?

Also, can someone give an example of a npm package that includes some code that has side effects (for example, some module `setWindowProp.js` that is setting some property on the global window object) and then if you set `sideEffect: false` in this package will cause consuming apps to skip that code, and if you set `sideEffects: ["setWindowProp.js"]` it will cause the consuming app to include that code in the app's bundle even if it is not imported directly

Thank you

**Comment by alexander-akait** [comment]
Please open an issue in the documentation repository and we will improve this.

**Reply by yishay-at-bay** [reply]
@alexander-akait thanks for responding
I opened an issue [here](https://github.com/webpack/webpack.js.org/issues/7414)
It would be great if you can still give here a small example with a npm package and a consumer app that demonstrates how the sideEffects flag can influence the inclusion/exclusion of some module with side effects

Thanks
